build.gradle

```diff
@@ -137,6 +140,36 @@ jar {
     }
 }
 
+// Do shadow relocations
+// (I can't figure out how to do this in a manual, sensible way, so--for
+// now--you will have to hunt down these dependencies manually and iteratively.
+// Sorry!  - Grissess)
+def shaded = [
+    "com.charleskorn.kaml",
+	"org.snakeyaml",
+].stream().collect()
```

Comment:
So.. this works? I find it surprising that selecting this from the implementation/library/shadow clause doesn't work.. hmmm.

Reply:
The best guess I have here is that the _automatic_ relocation is a bit overzealous and has a proclivity of selecting _too much_. By default, a ton of things are selected (as you may have noticed with the 92MB shadow jar containing most of MC's assets), and there's [no way to reset that list before adding to it again](https://discuss.gradle.org/t/sourcesets-how-can-i-include-exclude-and-include-again-in-sourcesets-in-that-order/5631/2). Thus, the current hacky solution uses the increased priority of `exclude` to add an (inverted) filter--indeed, a local run of `shadowJar` will print out virtually every package registered to `gradlew dependencies` (and some more unlisted dependencies!) as posited for inclusion in the shadow JAR. I'm just sorry I can't find a better way of dealing with it right now.